pandas/core/generic.py

```diff
@@ -5894,6 +5894,15 @@ def pipe(
         Use ``.pipe`` when chaining together functions that expect
         Series, DataFrames or GroupBy objects. Instead of writing
 
+        >>> df = pd.Dataframe(np.random.rand(2, 2))
+        >>> h = lambda df: [x.lower() for x in df.columns]
+        >>> def g(x, arg1):
+        ...     return x * arg1
+        >>> def func(x, arg2, arg3):
+        ...     return x * arg2 * arg3
+        >>> a = 1
+        >>> b = 2
+        >>> c = 3
         >>> func(g(h(df), arg1=a), arg2=b, arg3=c)  # doctest: +SKIP
```

Comment:
for this section, I think we should either:

- keep it vague (like it currently is) and `doctest: +skip` it
- or make it runnable (like you've done) and remove the `doctest: skip`

Looks like this doesn't quite run as it is - for example, the first line should be `pd.DataFrame`, rather than `pd.Dataframe`

Reply:
What I think would be really helpful here, besides passing the linting, is to find a better example that looks like real data. Personally, I think all these `g`, `h`... don't make it very clear how `.pipe` helps, since the original example is a bit cryptic.

@kkangs0226 what do you think? Maybe something like a small dataset with people and salaries, and have couple of functions that substract taxes (federal and state, or tax and national insurance). Or maybe a totally different example that don't make people feel sad ;)

Surely feel free to leave this out of this PR, we can create an issue for someone else to take care of if you prefer.

Reply:
I've added the example for the taxes, I think it makes the ```pipe``` functionality much more comprehensive!